observation/store.go

```diff
@@ -95,7 +95,7 @@ func (store *Store) SaveAll(observations []*Observation) ([]*Result, error) {
 			for instanceID := range instanceObservations {
 				store.reportError(instanceID, "observation batch insert failed", err)
 			}
-			return nil, err
+			continue
```

Comment:
if I'm reading this right, there is now **no circumstance** where the `conn.ExecNeo(...)` statement will stop the kafka message from being committed

i.e. kafka-retries will never happen, no matter the state of the database

sounds scary

Reply:
That is correct. Any issues from running a query will call the error reporter and mark the import as failed. We are consuming the message in these cases to prevent getting into a cycle of consuming the same message and it failing over again. We have upcoming work to improve the error handling and add retry logic 👍  I have added a comment that hopefully clarifies this.